src/lsst/rsp/client.py

```diff
@@ -13,20 +16,53 @@ class RSPClient(httpx.AsyncClient):
     """
 
     def __init__(
-        self,
-        service_endpoint: str,
+        self, service_endpoint: str, token: str | None = None
     ) -> None:
-        token = get_access_token()
+        # Accept a passed token.  This can be useful, in conjunction with
+        # forcing service_endpoint to be somewhere in some other RSP, for
+        # making cross-RSP calls.
+        if token is None:
+            token = get_access_token()
         jupyterlab_dir = get_runtime_mounts_dir()
         instance_url = (
             (jupyterlab_dir / "environment" / "EXTERNAL_INSTANCE_URL")
             .read_text()
             .strip()
         )
-        if instance_url.endswith("/") or service_endpoint.startswith("/"):
-            service_root = f"{instance_url}{service_endpoint}"
+        parsed_endpoint = urlparse(service_endpoint)
+        parsed_instance = urlparse(instance_url)
+        if parsed_endpoint.scheme:
+            # We expect people to be feeding this a client based on the
+            # results of Repertoire discovery, which will yield a full URL.
+            # That will have both a scheme and a netloc (in urlparse() terms).
+            #
+            # If the netloc doesn't match (possibly with an "nb" in
+            # front of it, which is true for Nublado with user domains
+            # enabled) then warn the user but keep going.  Likewise
+            # for the scheme, which must match exactly.  It's possible they
+            # are intentionally making a call to a different RSP instance.
+            if parsed_endpoint.scheme != parsed_instance.scheme or (
+                parsed_endpoint.netloc != parsed_instance.netloc
+                and f"nb.{parsed_endpoint.netloc}" != parsed_instance.netloc
+            ):
+                logger = logging.getLogger(__name__)
```

Comment:
Minor personal preference would be to include this at the top

Reply:
I think I'm going to keep it the way it is, so we match the common case first and short circuit the comparison.